app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/BaseViewModel.kt

```diff
@@ -92,7 +93,22 @@ abstract class BaseViewModel : ViewModel(), KoinComponent {
     /** Helper function to avoid needing downcast declarations for public MutableLiveData or LiveEvent */
     protected fun <T> LiveData<T>.postValue(value: T?) = (this as? MutableLiveData<T>)?.postValue(value) ?: run { Timber.w("[postValue] unable to postValue for $this") }
 
+    /**
+     *  Helper functions to get access down casted mutable SharedFlows
+     *    due to SharedFlow being covariant we must use templates with upper bounds to show type errors at build instead of run time.
+     */
+    protected suspend fun <T : Number?> SharedFlow<T>.emit(value: T) =
+        (this as? MutableSharedFlow<T>)?.emit(value) ?: run { Timber.w("[emitValue] unable to emit value for $this") }
+    protected suspend fun <T : CharSequence> SharedFlow<T>.emit(value: T) =
+        (this as? MutableSharedFlow<T>)?.emit(value) ?: run { Timber.w("[emitValue] unable to emit value for $this") }
+    protected suspend fun SharedFlow<Boolean>.emit(value: Boolean) =
+        (this as? MutableSharedFlow<Boolean>)?.emit(value) ?: run { Timber.w("[emitValue] unable to emit value for $this") }
+    protected suspend fun SharedFlow<Unit>.emit(value: Unit) =
+        (this as? MutableSharedFlow<Unit>)?.emit(value) ?: run { Timber.w("[emitValue] unable to emit value for $this") }
```

Comment:
@BottleRocket-Colin Thanks for getting to the bottom of the issue and creating an approach to resolve the the issue!

It looks like we _might_ have something coming up in Kotlin 1.7 that might obsolete the need for these types of boilerplate reducer functions by directly allowing separate `public` and `private` types for a property 🙏🏻 : https://youtrack.jetbrains.com/issue/KT-14663#focus=Comments-27-5961589.0-0